hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/store/NearCacheDataRecordStore.java

```diff
@@ -157,4 +167,18 @@ public int size() {
         return store.size();
     }
 
+    @Override
+    public void doExpiration() {
+        for (Map.Entry<K, NearCacheDataRecord> entry : store.entrySet()) {
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
We are checking interrupted status for every iteration, is this necessary ?
When do we need to interrupt this thread ? while destroying ?